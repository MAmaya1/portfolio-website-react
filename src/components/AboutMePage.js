import React from 'react';

// Import Footer

import Footer from './Footer';

// Import Images

import portrait from '../images/portrait.jpg'

const AboutMePage = props => {
    return (
        <div>
            <h1>About <span>Me</span></h1>
            <h2>Let me tell you a few things...</h2>
            <div className="about-info">
                <img src={portrait}/>
                <div className="bio">
                    <h3>BIO</h3>
                    <p>I am a React.js developer based in San Diego, and I have several years of experience as a highly-qualified project manager and customer service professional in the fields of multimedia and technology.</p>
                </div>
                <div className="info-box">
                    <h3>Technical Skills</h3>
                    <p>React, Redux, React Native, React Hooks, React Styled Components, React Testing Library, Axios, NodeJS, Express, Knex, Jest, LESS/SASS</p>
                </div>
                <div className="info-box">
                    <h3>Professional Skills</h3>
                    <p>Project Management, Leadership, Problem Solving, Employee Training, Task Delegation, Product Presentation, Time Management</p>
                </div>
                <div className="info-box">
                    <h3>Personal Interests</h3>
                    <p>Local Urban Planning, Community Activism, Architectural Design, Model Building, Writing, Animation, Film Production, and Travel.</p>
                </div>
            </div>
            <Footer />
        </div>
    )
}

export default AboutMePage;