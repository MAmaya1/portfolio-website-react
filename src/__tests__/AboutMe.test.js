import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// Import Component
import AboutMePage from '../components/AboutMePage';
import { get } from 'https';

// afterEach(cleanup);

describe('About Me component', () => {
    it('should render component without crashing', () => {
        render(<AboutMePage />);
    });

    it('should match Snapshot', () => {
        const wrapper = render(<AboutMePage />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render an "About Me" heading', () => {
        const { getByText } = render(<AboutMePage />);
        expect(getByText('About')).toBeInTheDocument();
    });

    it('should render a capitalized "BIO" subheading', () => {
        const { getByText } = render(<AboutMePage />);
        expect(getByText('BIO')).toBeInTheDocument();
    });

    it('should contain a "Technical Skills" section', () => {
        const { getByText } = render(<AboutMePage />);
        expect(getByText('Technical Skills')).toBeInTheDocument();
    });

    it('should contain a "Professional Skills" section', () => {
        const { getByText } = render(<AboutMePage />);
        expect(getByText('Professional Skills')).toBeInTheDocument();
    });

    it('should contain a "Personal Interests" section', () => {
        const { getByText } = render(<AboutMePage />);
        expect(getByText('Personal Interests')).toBeInTheDocument();
    });
})