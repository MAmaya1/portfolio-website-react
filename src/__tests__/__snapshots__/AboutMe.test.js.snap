// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Me component should match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxirZ cWwpYu"
      >
        <h1
          class="sc-AxiKw fXcBkv"
        >
          About 
          <span>
            Me
          </span>
        </h1>
        <h2
          class="sc-AxhCb fPoYls"
        >
          Let me tell you a few things...
        </h2>
        <div
          class="sc-AxhUy cfLSnK"
        >
          <img
            alt="Mario Amaya, React Developer"
            class="sc-AxgMl iJLcUn bio-pic"
            src="portrait.jpg"
          />
          <div
            class="sc-AxheI jVkjRL"
          >
            <h3>
              BIO
            </h3>
            <p>
              I am a React.js developer based in San Diego, and I have several years of experience as a highly-qualified project manager and customer service professional in the fields of multimedia and technology.
            </p>
          </div>
          <div
            class="sc-Axmtr bejfja box1"
          >
            <h3>
              Technical Skills
            </h3>
            <p>
              React, Redux, React Native, React Hooks, React Styled Components, React Testing Library, Axios, NodeJS, Express, Knex, Jest, LESS/SASS
            </p>
          </div>
          <div
            class="sc-Axmtr bejfja box2"
          >
            <h3>
              Professional Skills
            </h3>
            <p>
              Project Management, Leadership, Problem Solving, Employee Training, Task Delegation, Product Presentation, Time Management
            </p>
          </div>
          <div
            class="sc-Axmtr bejfja box3"
          >
            <h3>
              Personal Interests
            </h3>
            <p>
              Local Urban Planning, Community Activism, Architectural Design, Model Building, Writing, Animation, Film Production, and Travel.
            </p>
          </div>
        </div>
      </div>
      <footer
        class="sc-AxjAm hnGFdg"
      >
        <h5>
          Mario Amaya | 
          <span>
            React Developer
          </span>
        </h5>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxirZ cWwpYu"
    >
      <h1
        class="sc-AxiKw fXcBkv"
      >
        About 
        <span>
          Me
        </span>
      </h1>
      <h2
        class="sc-AxhCb fPoYls"
      >
        Let me tell you a few things...
      </h2>
      <div
        class="sc-AxhUy cfLSnK"
      >
        <img
          alt="Mario Amaya, React Developer"
          class="sc-AxgMl iJLcUn bio-pic"
          src="portrait.jpg"
        />
        <div
          class="sc-AxheI jVkjRL"
        >
          <h3>
            BIO
          </h3>
          <p>
            I am a React.js developer based in San Diego, and I have several years of experience as a highly-qualified project manager and customer service professional in the fields of multimedia and technology.
          </p>
        </div>
        <div
          class="sc-Axmtr bejfja box1"
        >
          <h3>
            Technical Skills
          </h3>
          <p>
            React, Redux, React Native, React Hooks, React Styled Components, React Testing Library, Axios, NodeJS, Express, Knex, Jest, LESS/SASS
          </p>
        </div>
        <div
          class="sc-Axmtr bejfja box2"
        >
          <h3>
            Professional Skills
          </h3>
          <p>
            Project Management, Leadership, Problem Solving, Employee Training, Task Delegation, Product Presentation, Time Management
          </p>
        </div>
        <div
          class="sc-Axmtr bejfja box3"
        >
          <h3>
            Personal Interests
          </h3>
          <p>
            Local Urban Planning, Community Activism, Architectural Design, Model Building, Writing, Animation, Film Production, and Travel.
          </p>
        </div>
      </div>
    </div>
    <footer
      class="sc-AxjAm hnGFdg"
    >
      <h5>
        Mario Amaya | 
        <span>
          React Developer
        </span>
      </h5>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
